# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/CrossShadowBoundaryRange.cpp
# Commit: 0825e43199d2
# Full Hash: 0825e43199d20888d87c3038413cdb652712a275
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2024-07-16 04:08:30
# Regressor Bug: 1907464
# File Overlap Count: 1
# Description:
#   Bug 1907464 - Clear the shadow-crossing selection when the ParentChainChanged notification is called on the common ancestor r=jjaschke,dom-core
#   
#   It's unclear from the spec about where the new selection should be,
#   so we just clear the selection at the moment.
#   
# ==============================================================================

diff -r 399b1edeca46 -r 0825e43199d2 dom/base/CrossShadowBoundaryRange.cpp
--- a/dom/base/CrossShadowBoundaryRange.cpp	Mon Jul 15 15:20:11 2024 +0000
+++ b/dom/base/CrossShadowBoundaryRange.cpp	Mon Jul 15 15:29:05 2024 +0000
@@ -243,4 +243,13 @@
                mOwner);
   }
 }
+
+// DOM mutation for shadow-crossing selection is not specified.
+// Spec issue: https://github.com/w3c/selection-api/issues/168
+void CrossShadowBoundaryRange::ParentChainChanged(nsIContent* aContent) {
+  MOZ_ASSERT(mCommonAncestor == aContent,
+             "Wrong ParentChainChanged notification");
+  MOZ_ASSERT(mOwner);
+  mOwner->ResetCrossShadowBoundaryRange();
+}
 }  // namespace mozilla::dom